src/index.ts

```diff
@@ -1,3 +1,111 @@
-export function init() {
-  return 'init...';
+import { IMetaQuerier, BasicMetaQuerier } from './util/metaquerier';
+import {
+  matchPattern,
+  shouldStartNewOne,
+  isDragStep,
+  shouldStopCurrentOne,
+  needCollect,
+} from './match-pattern';
+import { EventObserver } from './observer';
+import { Step, StepEvent } from './types';
+import { EventHandler } from './util/eventbus';
+
+type StepEventHandler = (step: Step) => void;
+
+export type RecorderOptions = {
+  doc: Document;
+  win: Window;
+  onEmit: StepEventHandler;
+  metaQuerier?: IMetaQuerier;
+  onMouseDown?: EventHandler;
+  onMouseUp?: EventHandler;
+  onClick?: EventHandler;
+  onMouseMove?: EventHandler;
+  onScroll?: EventHandler;
+  onKeyDown?: EventHandler;
+  onKeyPress?: EventHandler;
+  onKeyUp?: EventHandler;
+  onTextInput?: EventHandler;
+  onBlur?: EventHandler;
+};
+
+export class Recorder {
+  private eventObserver: EventObserver;
+
+  private currentTarget: HTMLElement | null = null;
+  private currentEvents: StepEvent[] = [];
+
+  private onEmit: StepEventHandler;
+  private metaQuerier: IMetaQuerier;
+
+  constructor(options: RecorderOptions) {
+    const { onEmit, metaQuerier, ...restOptions } = options;
+    this.onEmit = onEmit;
+    this.eventObserver = new EventObserver({
+      ...restOptions,
+      onEmit: this.eventHandler.bind(this),
+    });
+    this.metaQuerier = metaQuerier || new BasicMetaQuerier();
+  }
+
+  public start(): void {
+    this.eventObserver.start();
+  }
+
+  public suspend(): void {
```

Comment:
A `stop` function is needed. The expected behavior of the `stop` function is to stop all the event listeners and do not have any runtime cost anymore.

Reply:
And what's the use case of `suspend` when we have `stop`?

Reply:
suspend will stop sending event to event bus until you reactive the recorder
stop will destroy the observer and cannot reactive until you create a new observer.